src/Microsoft.AspNet.Mvc.Core/MvcCoreServiceCollectionExtensions.cs

```diff
@@ -48,59 +46,49 @@ internal static void AddMvcCoreServices(IServiceCollection services)
         {
             // Options
             //
-            TryAddMultiRegistrationService(
-                services,
+            services.TryAddEnumerable(
                 ServiceDescriptor.Transient<IConfigureOptions<MvcOptions>, CoreMvcOptionsSetup>());
 
             // Action Discovery
             //
             // These are consumed only when creating action descriptors, then they can be de-allocated
-            services.TryAdd(ServiceDescriptor.Transient<IAssemblyProvider, DefaultAssemblyProvider>());
-            services.TryAdd(ServiceDescriptor.Transient<IControllerTypeProvider, DefaultControllerTypeProvider>()); ;
-            TryAddMultiRegistrationService(
-                services,
+            services.TryAddTransient<IAssemblyProvider, DefaultAssemblyProvider>();
+            services.TryAddTransient<IControllerTypeProvider, DefaultControllerTypeProvider>();
+            services.TryAddEnumerable(
                 ServiceDescriptor.Transient<IApplicationModelProvider, DefaultApplicationModelProvider>());
-            TryAddMultiRegistrationService(
-                services,
+            services.TryAddEnumerable(
                 ServiceDescriptor.Transient<IActionDescriptorProvider, ControllerActionDescriptorProvider>());
-            services.TryAdd(ServiceDescriptor
-                .Singleton<IActionDescriptorsCollectionProvider, DefaultActionDescriptorsCollectionProvider>());
+            services.TryAddSingleton<IActionDescriptorsCollectionProvider, DefaultActionDescriptorsCollectionProvider>();
 
             // Action Selection
             //
-            services.TryAdd(ServiceDescriptor.Singleton<IActionSelector, DefaultActionSelector>());
+            services.TryAddSingleton<IActionSelector, DefaultActionSelector>();
             // Performs caching
```

Comment:
Getting rather cramped. Please add blank lines to make these blocks more readable.